engineering_tools/opt_carma_setup.bash

```diff
@@ -27,29 +68,44 @@ sudo groupadd --gid $GRP_ID carma # create the carma group if it does not alread
 USERNAME=$(whoami)
 sudo usermod -a -G $GRP_ID $USERNAME
 
-sudo mkdir -p /opt/carma/logs /opt/carma/.ros /opt/carma/maps /opt/carma/routes /opt/carma/yolo/darknet/data
-sudo chgrp -R $GRP_ID /opt/carma/
-sudo chmod 775 -R /opt/carma/
-sudo chmod 775 /opt/carma/logs /opt/carma/.ros
+mkdir -p /opt/carma/logs /opt/carma/.ros /opt/carma/maps /opt/carma/routes /opt/carma/yolo /opt/carma/data
 
-FILE_EXT=$RANDOM
-mkdir ~/carma_temp_$FILE_EXT
-cd ~/carma_temp_$FILE_EXT
+# Check if downloading maps is required
+if [ "$DOWNLOAD_MAPS" == "true" ]; then
+    FILE_EXT=$RANDOM
+    mkdir ~/carma_temp_$FILE_EXT
+    cd ~/carma_temp_$FILE_EXT
 
-git clone --branch carma-develop --depth 1 https://github.com/usdot-fhwa-stol/autoware.ai.git
-cp -R autoware.ai/core_perception/vision_darknet_detect/darknet/cfg/ /opt/carma/yolo/darknet/cfg/
-curl -o /opt/carma/yolo/darknet/data/yolov3.weights -L https://pjreddie.com/media/files/yolov3.weights
-chmod 775 /opt/carma/yolo/darknet/data/yolov3.weights
+    echo "Downloading example maps..."
+    curl -o /opt/carma/maps/pcd_map.pcd -L https://raw.githubusercontent.com/usdot-fhwa-stol/carma-config/develop/example_opt_carma/maps/base_map.pcd
 
-curl -o /opt/carma/maps/pcd_map.pcd -L https://raw.githubusercontent.com/usdot-fhwa-stol/carma-config/develop/example_opt_carma/maps/base_map.pcd
-chmod 775 /opt/carma/maps/pcd_map.pcd
+    curl -o /opt/carma/maps/vector_map.osm -L https://raw.githubusercontent.com/usdot-fhwa-stol/carma-platform/develop/route/resource/map/town01_vector_map_1.osm
 
-curl -o /opt/carma/maps/vector_map.osm -L https://raw.githubusercontent.com/usdot-fhwa-stol/carma-platform/develop/route/resource/map/town01_vector_map_1.osm
-chmod 775 /opt/carma/maps/vector_map.osm
+    curl -o /opt/carma/routes/Test_lanelet111_route_1.csv -L  https://raw.githubusercontent.com/usdot-fhwa-stol/carma-platform/develop/route/resource/route/Test_lanelet111_route_1.csv
 
-curl -o /opt/carma/routes/Test_lanelet111_route_1.csv -L  https://raw.githubusercontent.com/usdot-fhwa-stol/carma-platform/develop/route/resource/route/Test_lanelet111_route_1.csv
-chmod 775 /opt/carma/routes/Test_lanelet111_route_1.csv
+    # Clean up temporary files
+    rm -R ~/carma_temp_$FILE_EXT
+else
+    echo "Skipping map downloads..."
+fi
 
-ln -s "$1" /opt/carma/vehicle
+sudo chgrp -R $GRP_ID /opt/carma/
+sudo chmod 775 -R /opt/carma/
 
-sudo rm -R ~/carma_temp_$FILE_EXT
+# Link the provided vehicle calibration folder
+if [ -L /opt/carma/vehicle ]; then
+    sudo rm /opt/carma/vehicle
+fi
+
+ln -s "$CALIBRATION_FOLDER" /opt/carma/vehicle
+
+if [ "$ADD_SIM_FOLDER" == "true" ]; then
+    echo "Adding folders used for CDASim to be able to save logs"
+    mkdir -p /opt/carma/logs/carma_1 /opt/carma/logs/carma_2 /opt/carma-simulation/logs /opt/carma-simulation/scripts /opt/carma-streets/scripts
```

Comment:
nit: Manually adding log directories for each component seems like this will change pretty frequently. Since the permissions are already elevated, can we not let these directories be created on their own?

Reply:
These folders do not exist yet. Therefore, currently when the carma runs, it creates with permissions of root root. This is why carma keeps failing to write logs and won't start. If the folder exists with correct permissions, files can be written. 

it must be possible to make our docker configuration to write with correct permissions. But I just don't have enough bandwidth to investigate and fix.